streamparse/storm/bolt.py

```diff
@@ -300,10 +304,10 @@ def process_batch(self, key, tups):
     Note: Can be fractional to specify greater precision (e.g. 2.5).
     """
 
-    def __init__(self):
-        super(BatchingBolt, self).__init__()
+    def __init__(self, *args, **kwargs):
+        super(BatchingBolt, self).__init__(*args, **kwargs)
         self.exc_info = None
-        signal.signal(signal.SIGINT, self._handle_worker_exception)
+        signal.signal(signal.SIGUSR1, self._handle_worker_exception)
```

Comment:
Yeah, that makes a lot of sense.  I'm cool with that.  There's still the fact that a BatchingBolt has to exist on the main thread, or signal handlers don't attach properly, but I guess that hasn't been an issue for anyone yet.